packages/ics721/src/query.rs

```diff
@@ -88,6 +96,28 @@ pub fn query_nft_contract_for_class_id(
         .map(|e| e.map(|(_, v)| v.address))
 }
 
+pub fn query_get_nft_contract_for_class_id(
```

Comment:
Would change the name to reflect that it uses init2 (which might not be the actual address if wrong info is given).

Also why we even need it? we have the storage, if the contract doesn't exists in storage, means we didn't init it yet, and we can use init2 with the saved code in storage.